samcli/commands/init/__init__.py

```diff
@@ -126,12 +149,25 @@ def do_cli(
     # check for required parameters
     if location or (name and runtime and dependency_manager and app_template):
         # need to turn app_template into a location before we generate
-        extra_context = None
         if app_template:
             templates = InitTemplates(no_interactive, auto_clone)
             location = templates.location_from_app_template(runtime, dependency_manager, app_template)
             no_input = True
-            extra_context = {"project_name": name, "runtime": runtime}
+            default_context = {"project_name": name, "runtime": runtime}
```

Comment:
Can we actually pull this into another method? The method would handle all of the default context and merging of extra context, and return the new context value to pass to later calls.